ToDoApp/ToDoApp/ToDoListViewModel.swift

```diff
@@ -67,7 +67,6 @@ final class ToDoListViewModel: ObservableObject {
         if case .loading = state {
             return
         }
-        state = .loading
```

Comment:
In that case user will see empty screen on first launch if it takes long time to fetch todo items

Reply:
So should I still keep this line, and find a different way to avoid flickering when changing the task's status?

Reply:
yes, so basically why do we need the loading state? to show a loader when fetching tasks so that user would know that something is going on. also we want that the user would see that something is happening after saving new/edited tasks but we don't want to hide the list. so we need a way to toggle loading state while keeping the list on the screen

Reply:
So do you want to create a custom loading view (at the end of the article) in this article https://www.swiftbysundell.com/articles/handling-loading-states-in-swiftui/ when it's in loading state to still show the list? Or should I set up a condition so that the program will go to this line when it satisfies the condition?

Reply:
I have briefly come through the article and I haven't seen how it handles loading when the state was loaded previously. Can you tell me more?

Reply:
I saw in that, there's a section called custom loading view, so I wonder, do you want me to replace ProgressView() in case .loading to a custom loading view, where it will show the list?

Reply:
I think I don't get the idea